### PR TITLE
Add commenting to exported methods

### PR DESCRIPTION
**Pull request by itsmontoya**

### CODE REVIEW

File: terms.go [hunk 1]
```diff
@@ -82,19 +91,14 @@ func (e *Env) Model() G.Nodes {
...
 }
 
+// HintedModel will return the gorgonia.Nodes hinted associated with this environment
 func (e *Env) HintedModel(hint int) G.Nodes {
 	prealloc := make(G.Nodes, 0, hint)
 	e.hinted(prealloc)
 	return prealloc
 }
 
-func (e *Env) hinted(prealloc G.Nodes) {
```
Comment (chewxy) [inline terms.go:91]: Why is this removed? `HintedModel` calls `hinted`

Comment (itsmontoya) [reply terms.go]: It was moved to the top, that way private methods and public methods are grouped together :)
